lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -1101,6 +1108,26 @@ getUTxOsStatistics ctx (ApiT wid) = do
         W.listUtxoStatistics wrk wid
     return $ toApiUtxoStatistics stats
 
+getWalletKeyHash
+    :: forall ctx s t k n.
+        ( s ~ SeqState n k
+        , ctx ~ ApiLayer s t k
+        , k ~ ShelleyKey
+        )
+    => ctx
+    -> ApiT WalletId
+    -> ApiT DerivationIndex
+    -> Handler ApiVerificationKeyHash
+getWalletKeyHash ctx (ApiT wid) (ApiT (DerivationIndex index)) =
+    withWorkerCtx @_ @s @k ctx wid liftE liftE $ \wrk -> do
+        (cp, _, _) <- liftHandler $ W.readWallet @_ @s @k wrk wid
+        let accPub = getState cp ^. #externalPool . #accountPubKey
+        let (KeyHash bytes) =
+                hashKey $ deriveMultisigPublicKey
+                (liftXPub $ Shelley.getKey accPub)
+                (toEnum $ fromInteger $ toInteger index)
```

Comment:
That logic would better be located in `Cardano.Wallet.hs`. What about defining a function along those lines:

```hs
getWalletKeyHash 
    ::  ctx
    -> WalletId
    -> AccountingStyle
    -> Index 'Soft AddressK
    -> KeyHash
```

and re-use it in this layer?